lang/java/compiler/src/test/java/org/apache/avro/compiler/specific/TestSpecificCompiler.java

```diff
@@ -17,14 +17,19 @@
  */
 package org.apache.avro.compiler.specific;
 
-import static org.hamcrest.CoreMatchers.equalTo;
```

Comment:
IMO static imports are usually before the non-static ones. Please update your IDE settings to not do these changes automatically.

Reply:
I see both in the codebase (I checked classes I did not alter).

Reply:
It appears the default differs per IDE: https://checkstyle.sourceforge.io/checks/imports/importorder.html (Eclipse sorts static imports first, IntelliJ last). I did not find a definition in the codebase.

I can add it to `.editorconfig`, but only using two IntelliJ-specific properties `ij_java_imports_layout` and `ij_java_layout_static_imports_separately`. Shall I do that?

Reply:
Another way is to tell IDEA to not format the lines which are not edited

Reply:
It took a while, but I've delved into this a bit: most other source files use a "java first, then others and static last" approach. However, many tests use a "static first, then java, then others" approach. Also, many files contain missorted imports.

It's probably best to simply stick to a single, preferably common, standard.

For that, there are 4 major choices: **CheckStyle**, **Google**, **Eclipse**, and **JetBrains**, with the latter two modified to not allow * imports (this mostly affects the JetBrains style).

Using these code styles (and two custom styles inferred from the code) yield the following number of changed files:

| Style      | .editorconfig approximation                          | Files changed |
|------------|------------------------------------------------------|---------------|
| Custom 1   | `java.**,javax.**,\|,*,\|,$*`                        | 389 (51%)     |
| Eclipse    | `$*,\|,java.**,\|,javax.**,\|,org.**,\|,com.**,\|,*` | 391 (51%)     | 
| JetBrains  | `*,\|,java.**,javax.**,\|,$*`                        | 454 (60%)     |
| Custom 2   | `$*,\|,*,\|,java.**,javax.**`                        | 516 (68%)     |
| Google     | `$*,\|,*`                                            | 548 (72%)     |
| CheckStyle | `*,$*`                                               | 554 (73%)     |

As you can see from the number (and percentage) of changed files, if the codebase ever followed an import code style, this was a very long time ago.

Taking a different approach, finding static imports and JDK (java/javax) imports among all and 1st imports, does describe a pattern:
* there are 233 files with static imports, of which 230 (99%) sort them before non-static imports
* there are 635 files with JDK imports, of which 109 (17%) sort them before other (non-static) imports

Using these two metrics, we should:
* sort static imports before non-static imports
* sort java/javax imports after other imports

This yields the code style labelled "Custom 2" in the table above.

Reply:
Hey, thanks so much for the investigation!  Can we put this "import style" issue into a JIRA and do it separately!?

We've alway just ... well, never bothered imposing an import order, but I can see that it's just an unnecessary pain because devs expect autoformat to leave unmodified code.

Haha, just weighing in though: we've chosen spotless:apply to format our code in the checks, and my preference is to do *whatever* spotless does by default :D  That ends up being the Google style in the table!

Reply:
Definitely don't format all files in this PR! :-)
As I said earlier - IMO the best would be to not sort the imports at all. Just use the current style for the modified file.